Automattic/jetpack
# Docker: Introduce configuration file

#### Changes proposed in this Pull Request:

* Add a default config file
* Add support for "override" config, that is excluded from git
* Provide support for old (deprecated) `extras` and `volumes` compose files
* Add deprecation comments into `extras` and `volumes` sample files

#### Jetpack product discussion


n/a
#### Does this pull request change what data or activity we track or use?


n/a
#### Testing instructions:







* Make sure that your existing docker setup is still functional
* Make some changes to the configuration, run `jetpack docker stop`, and examine newly generated `compose-mappings.built.yml` and `compose-extras.built.yml`
	* Create a `.docker-env.json` in the project root. Define some mappings following `.docker-env-default.json` example. Add new properties and override some.
	* Add some docker configuration into `.docker-env.json`
	* Update `compose-volumes.yml`: add new mapping and/or override the existing one
	* Update `compose-extras.yml`: add a new entry, or override the existing one
